report/report.go

```diff
@@ -0,0 +1,12 @@
+package report
+
+import ()
+
+type ReportableElement interface {
+	GetHeaders() []string
```

Comment:
I feel like "key" and "value" imply a map. Here the "headers" and "value" can be calculated on the fly, and not necessarily stored anywhere